### PR TITLE
WT-2154 - Make btree dump safer

### PR DESCRIPTION
**Pull request by daveh86**

### CODE REVIEW

**Comment by daveh86** [comment]
@agorrod, when you get a moment would you mind taking a look?

**Comment by agorrod** [comment]
@daveh86 I don't see any comments. We'd talked about encapsulating all of the changes in `bt_debug.c` - but I see a change to `bt_vrfy.c`.

Could you switch the asserts from being `session->dhandle != NULL` to `S2BT_SAFE(session) != NULL`

**Comment by agorrod** [comment]
@daveh86 I'm not sure if you wanted another review for this. A couple more comments:

* Some of the comments appear incomplete.
* Re the comments referencing why a btree handle is in the parameter list. The should mention that the functions may be called from within a debugger - so setting a btree handle in calling code is not  practical.

**Comment by daveh86** [comment]
please retest this

**Comment by daveh86** [comment]
retest this please

**Comment by daveh86** [comment]
@agorrod, can you have a further look at this?

File: src/btree/bt_debug.c [hunk 1]
```diff
@@ -524,11 +547,16 @@ __wt_debug_page(WT_SESSION_IMPL *session, WT_PAGE *page, const char *ofile)
 
 /*
  * __debug_tree --
- *	Dump the in-memory information for a tree.
+ *	Dump the in-memory information for a tree. Takes an explicit btree
```
Comment (agorrod) [inline src/btree/bt_debug.c]: Why is this comment different to the two equivalent comments above?

File: src/btree/bt_debug.c [hunk 2]
```diff
@@ -524,11 +547,16 @@ __wt_debug_page(WT_SESSION_IMPL *session, WT_PAGE *page, const char *ofile)
...
+ *	as one may not be set on the session. This is often the case as this
+ *	function will be called from within a debugger, which makes setting a
```
Comment (agorrod) [inline src/btree/bt_debug.c]: This version of the comment reads: 
"Takes an explicit btree as one may not be set on the session."

The other version reads:
"Takes an explicit btree as an argument, as one may not yet be set on the session."

I also don't think the final sentence in the comment adds value - it's obvious from the implementation that it's using ```WT_WITH_BTREE```

**Comment by agorrod** [comment]
lgtm

**Pull request merged**